# PhD_Notes


## Guppy - Translating electrices signal into nucleotides

Required input: fast5 files and the configuration file that sets the basecalling parameters depending on which flowcell and library preparation kit was used to produce the data

```guppy_basecaller -c dna_r9.4.1_450bps_hac.cfg -x 'auto' --recursive -i <input fast5 dirsctory> -s <output directory>```

R10.4 paper commands: The generated raw Nanopore data were basecalled in super-accurate mode using Guppy v. 5.0.16 with the dna_r9.4.1_450bps_sup.cfg model for R9.4.1 and the dna_r10.4_e8.1_sup.cfg model for R10.4 chemistry

## Porechop - Adapter trimming

The known Nanopore adapters that Porechop looks for are defined in the adapters.py file

Basic adapter trimming: 
```porechop -i input_reads.fastq.gz -o output_reads.fastq.gz```


## NanoFilt - Read trimming and filtering

NanoFilt to trim and filter MinION reads; NanoFilt does not provide options for input or output files. Therefore we will use the two redirect operators “>” and “<“ to

```NanoFilt –l 500 --headcrop 10 < porechopped.fastq > nanofilt_trimmed.fastq```

R10.4 paper: reads with a lower length than 200 bp and a Phred quality score below 7 and 10 for R9.4.1 and R10.4 reads, respectively

## Flye assembler - de novo assembler

Flye is a de novo assembler for single-molecule sequencing reads, such as those produced by PacBio and Oxford Nanopore Technologies

```flye --nano-hq inputfile.fa --threads N --meta```

R10.4 Paper: --meta --nano-hq
For Q20 data, use a combination of --nano-hq and --read-error 0.03

## Minimap2 - Mapping/Polishing

Takes the original ONT reads as input and realigns them to the assembly generated by Flye

```minimap2 -ax asm5 ref.fa asm.fa > aln.sam```


